File: tf_subscribe.cpp
#include "ros/ros.h"
#include "tf2_msgs/TFMessage.h"

void tfCallback(const tf2_msgs::TFMessage::ConstPtr& msg)
{
  ROS_INFO("translation_x: [%f]", msg->transforms[1].transform.translation.x);
  ROS_INFO("translation_y: [%f]", msg->transforms[1].transform.translation.y);
  ROS_INFO("rotation_z: [%f]", msg->transforms[1].transform.rotation.z);
}

int main(int argc, char **argv)
{
 
  ros::init(argc, argv, "tf_subscribe");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe<tf2_msgs::TFMessage>("/tf", 1000,tfCallback);
  ros::spin();

  return 0;
}

File: tf_subscribe.py
#!/usr/bin/env python
import rospy
import tf2_msgs.msg

def callback(msg):
    frame_id = msg.transforms[0].header.frame_id
    if frame_id == 'map':
        rospy.loginfo("translation_x %f",msg.transforms[1].transform.translation.x)
        rospy.loginfo("translation_y %f",msg.transforms[1].transform.translation.y)
        rospy.loginfo("rotation_z %f",msg.transforms[1].transform.rotation.z)

def listener():
    rospy.init_node('tf_subscribe', anonymous=True)
    rospy.Subscriber("/tf", tf2_msgs.msg.TFMessage, callback)
    rospy.spin()

if __name__ == '__main__':
    listener()
    